JSIL/ILBlockTranslator.cs

```diff
@@ -1130,11 +1138,28 @@ bool forDynamic
             var translated = TranslateNode(node as dynamic);
 
             var statement = translated as JSStatement;
-            if (statement == null) {
-                var expression = (JSExpression)translated;
+            if (statement == null)
+            {
+                var expression = (JSExpression) translated;
 
                 if (expression != null)
+                {
+                    var ranges =
+                        node.GetSelfAndChildrenRecursive<ILExpression>(item => true).SelectMany(item => item.ILRanges);
```

Comment:
If you could do this without ```GetSelfAndChildrenRecursive``` that would be good. I forget if they expose a way to treewalk directly. ```GetSelfAndChildrenRecursive``` is one of the biggest timewasters in JSIL profiles and it allocates a lot; in this case we're just going to be throwing its result away to begin with.